cpp/tensorrt_llm/kernels/quantization.cuh

```diff
@@ -446,10 +448,10 @@ __device__ uint64_t cvt_warp_fp16_to_mxfp8(PackedVec<Type>& vec, uint8_t* SFout)
     // Write the SF to global memory (STG.8).
     __nv_fp8_e8m0 tmpSFVal;
     tmpSFVal.__x = __nv_cvt_float_to_e8m0(SFValue, __NV_SATFINITE, cudaRoundPosInf);
-    SFValue = static_cast<float>(tmpSFVal);
+    float SFValueNarrow = static_cast<float>(tmpSFVal);
     fp8SFVal = tmpSFVal.__x;
     // Get the output scale (reciprocal of the SFValue).
-    float outputScale = SFValue != 0.f ? reciprocal_approximate_ftz(SFValue) : 0.0f;
+    float outputScale = SFValue != 0.f ? reciprocal_approximate_ftz(SFValueNarrow) : 0.0f;
```

Comment:
@PerkzZheng fyi I observed that `__nv_cvt_float_to_e8m0` converts `0` -> `2^-127` so this condition was bypassed and nans were produced later